src/model/storage/session.model.ts

```diff
@@ -1,14 +1,18 @@
-import { Account, PrivateKey } from '../general.types'
+import { Address, Bytes } from '../general.types'
 import { Network } from './network.model'
 
-export interface KeyData {
-  privateKey: PrivateKey
+export interface KeyData<SeedType> {
+  seed: SeedType
   url: string
 }
 
-export interface Session {
-  username: string
+export interface Session<SeedType> {
+  username?: string
+  account?: string
```

Comment:
as I checked you use `username` for ens name and `account` for local username.
I suggest to use `ensUserName` and `localUserName` instead, because through the code it is a bit confusing